hack/release/common.sh

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+config(){
+  GITHUB_ACCOUNT="Azure"
+  RELEASE_ACR=${RELEASE_ACR:-ksnap.azurecr.io} # will always be ovreridden
+  RELEASE_REPO_ACR=${RELEASE_REPO_ACR:-${RELEASE_ACR}/public/aks/karpenter/}
+  RELEASE_REPO_MAR=mcr.microsoft.com/aks/karpenter
+  SNAPSHOT_ACR=${SNAPSHOT_ACR:-ksnap.azurecr.io}
+  SNAPSHOT_REPO_ACR=${SNAPSHOT_REPO_ACR:-${SNAPSHOT_ACR}/karpenter/snapshot/}
+
+  CURRENT_MAJOR_VERSION="0"
+  RELEASE_PLATFORM="--platform=linux/amd64,linux/arm64"
+
+  MAIN_GITHUB_ACCOUNT="Azure"
+  RELEASE_TYPE_STABLE="stable"
+  RELEASE_TYPE_SNAPSHOT="snapshot"
+}
+
+# versionData sets all the version properties for the passed release version. It sets the values
+# RELEASE_VERSION_MAJOR, RELEASE_VERSION_MINOR, and RELEASE_VERSION_PATCH to be used by other scripts
+versionData(){
+  local VERSION="$1"
+  local VERSION="${VERSION#[vV]}"
+  RELEASE_VERSION_MAJOR="${VERSION%%\.*}"
+  RELEASE_VERSION_MINOR="${VERSION#*.}"
+  RELEASE_VERSION_MINOR="${RELEASE_VERSION_MINOR%.*}"
+  RELEASE_VERSION_PATCH="${VERSION##*.}"
+  RELEASE_MINOR_VERSION="v${RELEASE_VERSION_MAJOR}.${RELEASE_VERSION_MINOR}"
+}
+
+snapshot() {
+  RELEASE_VERSION=$1
+  echo "Release Type: snapshot
+Release Version: ${RELEASE_VERSION}
+Commit: $(git rev-parse HEAD)
+Helm Chart Version $(helmChartVersion "$RELEASE_VERSION")"
+
+  authenticatePrivateRepo
+  buildImages "${SNAPSHOT_REPO_ACR}"
+  updateHelmChart
+  # not locking artifacts for snapshot releases
+  cosignImage "${CONTROLLER_IMG}"
+  cosignImage "${CONTROLLER_IMG_NAP}"
+  publishHelmChart "karpenter" "${RELEASE_VERSION}" "${SNAPSHOT_REPO_ACR}"
+  publishHelmChart "karpenter-crd" "${RELEASE_VERSION}" "${SNAPSHOT_REPO_ACR}"
+}
+
+release() {
+  RELEASE_VERSION=$1
+  echo "Release Type: stable
+Release Version: ${RELEASE_VERSION}
+Commit: $(git rev-parse HEAD)
+Helm Chart Version $(helmChartVersion "$RELEASE_VERSION")"
+
+  authenticatePrivateRepo
+  buildImages "${RELEASE_REPO_ACR}"
+  updateHelmChart "${RELEASE_REPO_MAR}"
+  lockImage "${IMG_REPOSITORY}" "${IMG_TAG}"
+  lockImage "${IMG_REPOSITORY}" "${IMG_TAG}-aks"
+  cosignImage "${CONTROLLER_IMG}"
+  cosignImage "${CONTROLLER_IMG_NAP}"
+  updateHelmChart
+  publishHelmChart "karpenter" "${RELEASE_VERSION}" "${RELEASE_REPO_ACR}"
+  publishHelmChart "karpenter-crd" "${RELEASE_VERSION}" "${RELEASE_REPO_ACR}"
+}
+
+authenticatePrivateRepo() {
+  az acr login -n "${SNAPSHOT_REPO_ACR}"
```

Comment:
nit: don't really like the mixing of naming here, as this is used within the `release` script, and here its using the `SNAPSHOT_REPO_ACR` env var

Reply:
Is this actually a bug?

Reply:
Please leaving a comment to explain if intentional and keeping it this way.

Reply:
It is a bug, fixing with proper ACR for release. (This is expected to go away for release, with auth switching to 1ES runner-based instead.)